ql/src/java/org/apache/hadoop/hive/ql/parse/GenTezUtils.java

```diff
@@ -130,8 +130,8 @@ public static ReduceWork createReduceWork(
       maxPartition = (maxPartition > maxReducers) ? maxReducers : maxPartition;
 
       // reduce only if the parameters are significant
-      if (minPartition < maxPartition &&
-          nReducers * minPartitionFactor >= 1.0) {
+      final int parallelismThreshold = context.conf.getIntVar(HiveConf.ConfVars.TEZ_AUTO_REDUCER_PARALLELISM_THRESHOLD);
+      if (minPartition < maxPartition && maxPartition >= parallelismThreshold) {
```

Comment:
This looks like a behaviour change for the existing clients, if someone has configured ``minPartitionFactor`` according to his need, post your change, his results would change.

Reply:
Let me think of that point and an elegant way for 1 day. I understand we would like to satisfy the following conditions.
- We sometimes want to turn on auto reducer parallelism when `nReducers * minPartitionFactor` is less than 1.0 but the potential reduction is not trivial
- We should keep the original behavior by default

Reply:
more inclined towards option:2, keeps the current formula, just makes the comparison value configurable

Reply:
Thanks. It is also aligns with my current feeling. Let me try.

Reply:
@ayushtkn I added a commit and updated the test results in the description.
https://github.com/apache/hive/pull/4432/commits/851f91acc32be4e475bb2b40ffcca227aa6416a7